src/execution/execute.ts

```diff
@@ -139,9 +150,45 @@ export interface FormattedExecutionResult<
 > {
   errors?: ReadonlyArray<GraphQLFormattedError>;
   data?: TData | null;
+  hasNext?: boolean;
+  extensions?: TExtensions;
+  incremental?: ReadonlyArray<IncrementalResult>;
```

Comment:
`IncrementResult.errors` is a `ReadonlyArray<GraphQLError>` (not `ReadonlyArray<GraphQLFormattedError>`) so that's not a good fit for `FormattedExecutionResult`. I do find FormattedExecutionResult to be a useful type so that suggests we need Formatted versions of the Incremental types.